.eslintrc.js

```diff
@@ -22,78 +56,393 @@ module.exports = {
+    'import/no-named-as-default': 'off',
+    'import/no-webpack-loader-syntax': 'off',
+    'import/prefer-default-export': 'off', // Would be nice on react components?
+    'jsx-a11y/label-has-for': 'off',
+    'max-lines': 'off',
+    'max-lines-per-function': 'off',
+    'max-params': 'off',
+    'max-statements': 'off',
+    'max-statements-per-line': 'off',
+    'multiline-comment-style': 'off',
+    'newline-per-chained-call': 'off',
+    'no-confusing-arrow': 'off',
+    'no-inline-comments': 'off',
+    'no-process-exit': 'off',
+    'no-restricted-syntax': 'off',
+    'no-ternary': 'off',
+    'no-undefined': 'off',
+    'no-underscore-dangle': ['off', { allow: ['_display', '_links'] }],
+    'no-use-before-define': 'off', // Breaks on react hooks
     'prefer-destructuring': 'off',
-    'react/jsx-fragments': ['warn', 'element'],
+    'operator-linebreak': 'off',
+    'promise/no-native': 'off',
+    'react/destructuring-assignment': 'off',
+    'react/forbid-prop-types': 'off',
+    'react/jsx-closing-tag-location': 'off',
+    'react/jsx-curly-newline': 'off',
+    'react/jsx-handler-names': 'off',
+    'react/jsx-max-depth': 'off',
+    'react/jsx-max-props-per-line': 'off',
+    'react/jsx-no-literals': 'off',
+    'react/jsx-no-target-blank': 'off',
     'react/jsx-one-expression-per-line': 'off',
-    'react/jsx-indent': ['error', 2, { checkAttributes: true }],
-    semi: 'error',
-  },
-  settings: {
-    'import/resolver': {
-      node: {
-        extensions: ['.js', '.jsx', '.react.js'],
-        moduleDirectory: ['node_modules', './src'],
-      },
-    },
+    'react/jsx-props-no-spreading': 'off',
+    'react/jsx-sort-default-props': 'off',
+    'react/jsx-sort-props': 'off',
+    'react/jsx-wrap-multilines': 'off',
+    'react/no-multi-comp': 'off',
+    'react/no-set-state': 'off',
+    'react/require-default-props': 'off',
+    'react/require-extension': 'off',
+    'react/self-closing-comp': 'off',
+    'react/sort-comp': 'off',
+    'react/sort-prop-types': 'off',
+    'require-yield': 'off',
+    'sonarjs/cognitive-complexity': 'off',
+    'sort-imports': 'off',
+    'sort-keys': 'off',
+    'unicorn/consistent-function-scoping': 'off', // Breaks on react hooks
+    'unicorn/explicit-length-check': 'off',
+    'unicorn/filename-case': 'off',
+    'unicorn/new-for-builtins': 'off',
+    'unicorn/no-fn-reference-in-iterator': 'off',
+    'unicorn/no-nested-ternary': 'off',
+    'unicorn/no-null': 'off',
+    'unicorn/no-process-exit': 'off',
+    'unicorn/no-reduce': 'off',
+    'unicorn/no-useless-undefined': 'off',
+    'unicorn/prefer-node-append': 'off',
+    'unicorn/prefer-node-remove': 'off',
+    'unicorn/prefer-query-selector': 'off',
+    'unicorn/prefer-string-slice': 'off',
+
+    // disabled rules - related code should be refactored to functional components
+    'no-invalid-this': 'off',
+    'class-methods-use-this': 'off',
+    'react/no-did-mount-set-state': 'off',
```

Comment:
This is considered to be an anti-pattern. What is the impact of turning this on?

Reply:
The reason is that we still have class components triggered by those rules which require manual intervention to fix.